lib/project-data.ts

```diff
@@ -47,46 +49,131 @@ export class ProjectData implements IProjectData {
 
 	public initializeProjectData(projectDir?: string): void {
 		projectDir = projectDir || this.$projectHelper.projectDir;
+
 		// If no project found, projectDir should be null
 		if (projectDir) {
-			const projectFilePath = path.join(projectDir, this.$staticConfig.PROJECT_FILE_NAME);
-			let data: any = null;
+			const projectFilePath = this.getProjectFilePath(projectDir);
 
 			if (this.$fs.exists(projectFilePath)) {
-				let fileContent: any = null;
-				try {
-					fileContent = this.$fs.readJson(projectFilePath);
-					data = fileContent[this.$staticConfig.CLIENT_NAME_KEY_IN_PROJECT_FILE];
-				} catch (err) {
-					this.$errors.failWithoutHelp(`The project file ${this.projectFilePath} is corrupted. ${EOL}` +
-						`Consider restoring an earlier version from your source control or backup.${EOL}` +
-						`Additional technical info: ${err.toString()}`);
-				}
-
-				if (data) {
-					this.projectDir = projectDir;
-					this.projectName = this.$projectHelper.sanitizeName(path.basename(projectDir));
-					this.platformsDir = path.join(projectDir, constants.PLATFORMS_DIR_NAME);
-					this.projectFilePath = projectFilePath;
-					this.appDirectoryPath = path.join(projectDir, constants.APP_FOLDER_NAME);
-					this.appResourcesDirectoryPath = path.join(projectDir, constants.APP_FOLDER_NAME, constants.APP_RESOURCES_FOLDER_NAME);
-					this.projectId = data.id;
-					this.dependencies = fileContent.dependencies;
-					this.devDependencies = fileContent.devDependencies;
-					this.projectType = this.getProjectType();
-
-					return;
-				}
+				let packageJsonContent: any = null;
+				packageJsonContent = this.$fs.readText(projectFilePath);
+				const nsConfigContent: any = this.getNsConfigContent(projectDir);
+
+				this.initializeProjectDataFromContent(packageJsonContent, nsConfigContent, projectDir);
 			}
+
+			return;
+		}
+
+		this.errorInvalidProject(projectDir);
+	}
+
+	public initializeProjectDataFromContent(packageJsonContent: string, nsconfigContent: string, projectDir?: string): void {
+		projectDir = projectDir || this.$projectHelper.projectDir || "";
+		const projectFilePath = this.getProjectFilePath(projectDir);
+		// If no project found, projectDir should be null
+		let nsData: any = null;
+		let nsConfig: any = null;
```

Comment:
we definitely need an interface for nsConfig and the data that can be included in it. Can you please add a new one and use it as a type here instead of `any`.